File: leetcode/羅馬數字轉換阿拉伯數字.c
int romanToInt(char* s) {
    int num = 0;
    for(int i = 0; i < strlen(s); i++){
        int change = 0;
        switch(s[i]){
            case 'I': change = 1; break;
            case 'V': change = 5; break;
            case 'X': change = 10; break;
            case 'L': change = 50; break;
            case 'C': change = 100; break;
            case 'D': change = 500; break;
            case 'M': change = 1000; break;
        }

		if(i != strlen(s) - 1){
            if (s[i] == 'I' && s[i + 1] == 'V') { change = 4; i++; }
            if (s[i] == 'I' && s[i + 1] == 'X') { change = 9; i++; }
            if (s[i] == 'X' && s[i + 1] == 'L') { change = 40; i++; }
            if (s[i] == 'X' && s[i + 1] == 'C') { change = 90; i++; }
            if (s[i] == 'C' && s[i + 1] == 'D') { change = 400; i++; }
            if (s[i] == 'C' && s[i + 1] == 'M') { change = 900; i++; }
        }


        num += change;
    }

    return num;
}

File: Bypyhton/羅馬數字轉換阿拉伯數字.py
def romanToInt(s: str) -> int:
    num = 0
    i = 0
    while i < len(s):
        change = 0
        if s[i] == 'I': change = 1
        elif s[i] == 'V': change = 5
        elif s[i] == 'X': change = 10
        elif s[i] == 'L': change = 50
        elif s[i] == 'C': change = 100
        elif s[i] == 'D': change = 500
        elif s[i] == 'M': change = 1000

        if i != len(s) - 1:
            if s[i] == 'I' and s[i + 1] == 'V': change = 4; i += 1
            elif s[i] == 'I' and s[i + 1] == 'X': change = 9; i += 1
            elif s[i] == 'X' and s[i + 1] == 'L': change = 40; i += 1
            elif s[i] == 'X' and s[i + 1] == 'C': change = 90; i += 1
            elif s[i] == 'C' and s[i + 1] == 'D': change = 400; i += 1
            elif s[i] == 'C' and s[i + 1] == 'M': change = 900; i += 1

        num += change
        i += 1

    return num
